extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java

```diff
@@ -195,6 +196,20 @@ public boolean isValid(Connection connection) {
             poolConfiguration.maxLifetime(dataSourceRuntimeConfig.maxLifetime.get());
         }
 
+        // CloudSQL config
+        if (dataSourceRuntimeConfig.useGcp) {
+            disableSslSupport();
+            String cloudSqlInstance = dataSourceRuntimeConfig.cloudSqlInstance.orElseThrow(
+                    () -> new RuntimeException("Cloud sql instance property is mandatory to use Gcp Cloudsql"));
+            if (cloudSqlInstance.split(":").length != 3) {
+                throw new RuntimeException("Cloud sql instance should match the pattern project-id:zone:cloudSqlInstance");
+            }
+            agroalConnectionFactoryConfigurationSupplier.jdbcUrl(
+                    MessageFormat.format("{0}?socketFactory=com.google.cloud.sql.postgres.SocketFactory&cloudSqlInstance={1}",
```

Comment:
Maybe we should use the `agroalConnectionFactoryConfigurationSupplier.jdbcProperty(<property_name>, <value>) ` method here ?

Reply:
I don't think it works